rali/rali/source/tf_meta_data_reader.cpp

```diff
@@ -90,24 +90,16 @@ void TFMetaDataReader::read_record(std::ifstream &file_contents, uint file_size,
 {
     // std::cerr << "The user_label_key is " << user_label_key << ", and the user_filename_key is " << user_filename_key << "\n";
     uint length;
-    length = file_contents.tellg();
-    std::string temp;
-    size_t uint64_size, uint32_size;
     uint64_t data_length;
     uint32_t length_crc, data_crc;
-    uint64_size = sizeof(uint64_t); 
-    uint32_size = sizeof(uint32_t); 
-    char * header_length = new char [uint64_size];
-    char * header_crc = new char [uint32_size];
-    char * footer_crc = new char [uint32_size];
-    file_contents.read(header_length, uint64_size);
+
+    length = file_contents.tellg();
+    file_contents.read((char *)&data_length, sizeof(data_length));
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6366586)